src/prefect/orion/database/migrations/versions/postgresql/2023_01_31_110543_f98ae6d8e2cc_work_queue_data_migration.py

```diff
@@ -0,0 +1,189 @@
+    ).fetchone()[0]
+
+    default_queue = connection.execute(
+        sa.text("SELECT id FROM work_queue WHERE name = 'default'")
+    ).fetchone()
+
+    if not default_queue:
+        connection.execute(
+            sa.text(
+                f"INSERT INTO work_queue (name, work_pool_id) VALUES ('default', :default_pool_id)"
+            ).params({"default_pool_id": default_pool_id}),
+        )
+
+    connection.execute(
+        sa.text(
+            "UPDATE work_queue SET work_pool_id = :default_pool_id WHERE work_pool_id IS NULL"
+        ).params({"default_pool_id": default_pool_id}),
+    )
+
+    default_queue_id = connection.execute(
+        sa.text(
+            "SELECT id FROM work_queue WHERE name = 'default' and work_pool_id = :default_pool_id"
+        ).params({"default_pool_id": default_pool_id}),
+    ).fetchone()[0]
+
+    connection.execute(
+        sa.text(
+            "UPDATE work_pool SET default_queue_id = :default_queue_id WHERE id = :default_pool_id"
+        ).params(
+            {"default_pool_id": default_pool_id, "default_queue_id": default_queue_id}
+        ),
+    )
+
+    # Set priority on all queues and update flow runs and deployments
+    queue_rows = connection.execute(
+        sa.text(
+            "SELECT id, name FROM work_queue WHERE work_pool_id = :default_pool_id"
+        ).params({"default_pool_id": default_pool_id}),
+    ).fetchall()
+
+    with op.get_context().autocommit_block():
+        for enumeration, row in enumerate(queue_rows):
+            connection.execute(
+                sa.text(
+                    "UPDATE work_queue SET priority = :priority WHERE id = :id"
+                ).params({"priority": enumeration + 1, "id": row[0]}),
+            )
+
+            batch_size = 250
+
+            while True:
+                result = connection.execute(
+                    sa.text(
+                        """
+                        UPDATE flow_run 
+                        SET work_queue_id=:id 
+                        WHERE flow_run.id in (
+                            SELECT id 
+                            FROM flow_run 
+                            WHERE flow_run.work_queue_id IS NULL and flow_run.work_queue_name=:name 
+                            LIMIT :batch_size
+                        )
+                        """
+                    ).params({"id": row[0], "name": row[1], "batch_size": batch_size}),
+                )
+                if result.rowcount <= batch_size:
+                    break
+
+            while True:
+                result = connection.execute(
+                    sa.text(
+                        """
+                        UPDATE deployment 
+                        SET work_queue_id=:id 
+                        WHERE deployment.id in (
+                            SELECT id 
+                            FROM deployment 
+                            WHERE deployment.work_queue_id IS NULL and deployment.work_queue_name=:name 
```

Comment:
We might want to index `work_queue_name` indexed here, we can add a temporary index just for the migration which gets cleaned up later

Reply:
Added temporary indexes for the migration in [f42496b](https://github.com/PrefectHQ/prefect/pull/8327/commits/f42496bd958798d6b980c009470d9e49c393f0fb)